### DISCUSSION TITLE
Custom Migration / Add Trigger

### DISCUSSION THREAD

**Discussion by ainsleyclark** [payloadcms/payload#8649] [Q&A]
Hi there,

I'm trying to add a custom trigger to the database:

```sql
CREATE TRIGGER update_location_point_trigger
	BEFORE INSERT OR UPDATE
	ON places
	FOR EACH ROW
	WHEN (NEW.latitude IS NOT NULL AND NEW.longitude IS NOT NULL)
EXECUTE FUNCTION update_location_point();
```

I'm currently doing this in a seed script after all of the tables have been made.

But is it possible to add this to a custom migration? I.e is it possible to create a migration in the migration folder and have Payload latch on to it?

Alternatively I did try adding this to `afterSchemaInit` but the tables weren't created by then, is there a hook for when the tables are created?

**Comment by r1tsuu** [comment]
That's possible, create a new, empty migration with `payload migrate:create add_trigger` and put it there!

>, is there a hook for when the tables are created?

You can do this in `onInit`. They should be created there, something like `payload.adapter.drizzle.execute(sql.raw("<code>"))` should work. I'd go here with a migration though, but this approach can work with dev schema push. If you're going with this, make sure you don't execute it when the trigger already exists, possibly with this:
```sql
    SELECT EXISTS (
      SELECT 1
      FROM pg_trigger
      WHERE tgname = 'update_location_point_trigger'
    );
```

**Comment by ainsleyclark** [comment]
Thanks for your tips @r1tsuu! As I'm not using `db.push` the migration route is proving a bit tricky, as in dev I'm continually updating the migration dir so I have to rework `migrations/index.ts` every time I create a migration (which is often).

As for toe `onInit` I did give it a go but it came back with the tables were not created which I found a bit strange.
